src/operator/contrib/ngraph-inl.h

```diff
@@ -0,0 +1,148 @@
+
+class SgNgraphSelector : public SubgraphSelector {
+ public:
+  explicit SgNgraphSelector(ngraph_bridge::Compiler *compiler)
+      : compiler_(compiler), valid(compiler_->get_node_map().size() > 0) {}
+
+  bool Select(const nnvm::Node &n) override { return is_node_selected(n); }
+
+  bool SelectInput(const nnvm::Node &n, const nnvm::Node &new_node) override {
+    return is_node_selected(n, &new_node);
+  }
+
+  bool SelectOutput(const nnvm::Node &n, const nnvm::Node &new_node) override {
+    return is_node_selected(n, &new_node);
+  }
+  std::vector<nnvm::Node *> Filter(
+      const std::vector<nnvm::Node *> &candidates) {
+    if (candidates.size() == 1 && candidates[0]->inputs.size() == 0) {
+      return std::vector<nnvm::Node *>();
+    } else {
+      return candidates;
+    }
+  }
+
+ private:
+  ngraph_bridge::Compiler *compiler_;
+  const bool valid;
+  ngraph_bridge::NodePtr get_node(const nnvm::Node *n) {
+    if (n) {
+      auto &entry_map = compiler_->get_ngraph().entry_map_;
+      ngraph_bridge::MapEntry tmp{compiler_->get_node_map().at(n).get(), 0};
+      if (entry_map.count(tmp)) {
+        return entry_map[tmp];
+      }
+    }
+    return nullptr;
+  }
+  bool is_node_selected(const nnvm::Node &n, const nnvm::Node *next = nullptr) {
+    bool selected = false;
+    if (!valid) return selected;
+
+    auto nn = get_node(&n);
+    auto nnext = get_node(next);
+
+    selected = nn && nn->in_ngraph_;
+    if (next) {
+      selected =
+          selected && nnext->in_ngraph_ && nn->subgraph_ == nnext->subgraph_;
+    }
+    return selected;
+  }
+};
+
+class SgNgraphProperty : public SubgraphProperty {
+ public:
+  static SubgraphPropertyPtr Create() {
+    return std::make_shared<SgNgraphProperty>();
+    if (ngraph_backend != 0 && ngraph_bridge::ngraph_log_verbose_detail) {
+      std::cout
+          << "NGRAPH_BRIDGE: WARNING, failed to set MXNET_SUBGRAPH_BACKEND"
+          << std::endl;
+    }
+  }
+
+  bool NeedGraphAttrs() const override { return true; }
+  nnvm::NodePtr CreateSubgraphNode(
+      const nnvm::Symbol &sym, const int subgraph_id = 0) const override {
+    nnvm::NodePtr n = nnvm::Node::Create();
+    n->attrs.op = Op::Get("_ngraph_subgraph_op");
+    n->attrs.name = "_ngraph_subgraph_op" + std::to_string(subgraph_id);
+    n->attrs.subgraphs.push_back(std::make_shared<nnvm::Symbol>(sym));
+    return n;
+  }
+
+  nnvm::NodePtr CreateSubgraphNode(
+      const nnvm::Graph &sg, const int subgraph_id = 0) const override {
+    nnvm::Symbol sym;
+    sym.outputs = sg.outputs;
+    auto n = CreateSubgraphNode(sym, subgraph_id);
+
+    auto compiler = std::make_shared<ngraph_bridge::Compiler>(sg);
+    compiler->GetNgraph();
+    n->attrs.parsed = compiler;
+    return n;
+  }
+
+  SubgraphSelectorPtr CreateSubgraphSelector() const override {
+    if (!compiler_) {
+      auto &orig_graph = GetAttr<nnvm::Graph>("graph");
+      compiler_ = std::make_shared<ngraph_bridge::Compiler>(orig_graph, true);
+    }
```

Comment:
why not create `compiler` in a constructor?

Reply:
The compiler object requires an NNVM graph during construction, it's specific to an NNVM graph.